### DISCUSSION TITLE
InvalidArgument when selecting table_schema from information_schema.columns

### DISCUSSION THREAD

**Discussion by skeswa** [planetscale/discussion#478] [Bug Reports]
Hey there folks,

I'm getting a super weird issue where
```
select table_schema, table_name, ordinal_position from information_schema.columns where table_schema!='' limit 1;
```

works fine but then
```
select table_schema, table_name, ordinal_position from information_schema.columns where table_schema='' limit 1;
```

errors with the following:
```
ERROR 1105 (HY000): target: REDACTED.-.primary: vttablet: rpc error: code = InvalidArgument desc = missing bind var __vtschemaname (CallerID: planetscale-admin)
```

Context: a query similar to this is executed by [`drizzle-kit push:mysql`](https://github.com/drizzle-team/drizzle-orm). I also filed an issue with them, but this seems to be a Planetscale problem.

**Comment by orware** [comment]
Hello @skeswa,

I appreciate you reporting this one! I've been able to confirm it as well using the same examples you provided and it does seem to be a bit of a head scratcher.

For what it's worth, testing this from a local MySQL 8 instance shows the second query returning an empty result set and that's what I would expect from the PlanetScale database too, so I'm unsure if there are situations where Drizzle would expect that query to return any results?

This would likely require a fix from our Vitess team so I'll make sure it gets reported so it can be addressed, but just be aware that it may take some time.

**Reply by skeswa** [reply]
I don't really have context for _why_ `drizzle-kit` filters for a `table_schema` that is the empty string, but I suspect it might be related to how they store migrations.

Thanks for the super prompt response on this one! It is quite an exotic issue. I hope the drizzle folks can add an "if planetscale then use a different query" provision to their migration logic.

Cheers.

**Reply by orware** [reply]
Hello @skeswa,

It's been a while since my last update but I heard back from our Vitess team today that a public fix within Vitess has been created for the issue:
https://github.com/vitessio/vitess/issues/15902

These fixes are typically backported into the versions of Vitess we utilize within PlanetScale within a few weeks so I would expect this issue to go away in the near future!
